src/main/java/org/mule/extension/http/internal/request/HttpPollingSource.java

```diff
@@ -0,0 +1,186 @@
+import org.mule.extension.http.api.request.validator.ResponseValidator;
+import org.mule.extension.http.api.request.validator.SuccessStatusCodeValidator;
+import org.mule.extension.http.internal.HttpMetadataResolver;
+import org.mule.extension.http.internal.request.client.HttpExtensionClient;
+import org.mule.runtime.api.connection.ConnectionProvider;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.api.scheduler.Scheduler;
+import org.mule.runtime.api.scheduler.SchedulerService;
+import org.mule.runtime.api.transformation.TransformationService;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.util.IOUtils;
+import org.mule.runtime.extension.api.annotation.Alias;
+import org.mule.runtime.extension.api.annotation.Streaming;
+import org.mule.runtime.extension.api.annotation.metadata.MetadataScope;
+import org.mule.runtime.extension.api.annotation.param.Connection;
+import org.mule.runtime.extension.api.annotation.param.Config;
+import org.mule.runtime.extension.api.annotation.param.MediaType;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+import org.mule.runtime.extension.api.annotation.param.Parameter;
+import org.mule.runtime.extension.api.annotation.param.ParameterGroup;
+import org.mule.runtime.extension.api.annotation.param.display.Example;
+import org.mule.runtime.extension.api.annotation.param.display.Placement;
+import org.mule.runtime.extension.api.annotation.source.BackPressure;
+import org.mule.runtime.extension.api.runtime.operation.Result;
+import org.mule.runtime.extension.api.runtime.process.CompletionCallback;
+import org.mule.runtime.extension.api.runtime.source.BackPressureMode;
+import org.mule.runtime.extension.api.runtime.source.PollContext;
+import org.mule.runtime.extension.api.runtime.source.PollingSource;
+import org.mule.runtime.extension.api.runtime.source.SourceCallbackContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.List;
+
+@Alias("pollingSource")
+@MediaType(value = ANY, strict = false)
+@MetadataScope(outputResolver = HttpMetadataResolver.class)
+@Streaming
+@BackPressure(defaultMode = BackPressureMode.FAIL, supportedModes = {BackPressureMode.FAIL})
+public class HttpPollingSource extends PollingSource<String, HttpResponseAttributes> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(HttpPollingSource.class);
+
+  @Connection
+  private ConnectionProvider<HttpExtensionClient> clientProvider;
+
+  @Inject
+  private SchedulerService schedulerService;
+  private Scheduler scheduler;
+  @Inject
+  private TransformationService transformationService;
+  @Inject
+  @Named("http.request.fixedHeadersRegistry")
+  private HashMap<String, List<String>> injectedHeaders;
```

Comment:
shoudln't this be a MultiMap instead?

Reply:
This is exactly how it is for http requester: https://github.com/mulesoft/mule-http-connector/blob/master/src/main/java/org/mule/extension/http/internal/request/HttpRequestOperations.java#L84